tdrs-backend/tdpservice/search_indexes/parsers/tanf_parser.py

```diff
@@ -95,3 +100,52 @@ def parse(datafile):
         else:
             logger.warn("Parsing for type %s not yet implemented", record_type)
             continue
+
+def run_validation(family_case_schema, t1):
+    """Run validation on the datafile."""
+    cat2_errors = validate(family_case_schema, t1, 'cat2_conditions', validate_cat2)
+
+    logger.warn(f'There are {len(cat2_errors)} cat2 errors:')
+    for error in cat2_errors:
+        logger.warn(error)
+
+    cat3_errors = validate(family_case_schema, t1, 'cat3_conditions', validate_cat3)
+
+    logger.warn(f'There are {len(cat3_errors)} cat3 errors:')
+    for error in cat3_errors:
+        logger.warn(error)
+
+
+def validate(schema, model_obj, category, validator):
+    """Validate the datafile.
+
+    This function takes a parsing schema, which holds the fields and conditions,
+    A Django model obj, a category string that corresponds to a dictionary on the schema,
+    and a validator function.
+
+    For each row on the schema the field, condition, and value are taken and run through the
+    validator function, along with the model_obj.
+
+    If there are errors they are returned.
+    """
+    errors = []
+    for field in schema:
+        name = field['description']
+        if name == 'BLANK':
+            continue
+        value = getattr(model_obj, name)
+        category_conditions = field[category]
+        category_errors = None
+        if category_conditions != {}:
+            if 'custom' in category_conditions:
+                for custom_validator in category_conditions['custom']:
+                    category_errors = custom_validator(model_obj)
+                    if len(category_errors) > 0:
```

Comment:
They are equivalent. I prefer the above way because it is more descriptive of what you are checking for. The shortened version can imply that the variable is a boolean.

Reply:
`dict.get('key')` handles exceptions thrown if the key doesn't exist. it also lets you specify a default value; i.e., `dict.get('key', 'default')`. if you're sure the key exists and the value is acceptable, there's no difference